pkg/tsdb/prometheus/prometheus.go

```diff
@@ -283,13 +291,20 @@ func (s *Service) parseQuery(queryContext *backend.QueryDataRequest, dsInfo *Dat
 		expr = strings.ReplaceAll(expr, "$__range", strconv.FormatInt(rangeS, 10)+"s")
 		expr = strings.ReplaceAll(expr, "$__rate_interval", intervalv2.FormatDuration(calculateRateInterval(interval, dsInfo.TimeInterval, s.intervalCalculator)))
 
+		queryType := Range
```

Comment:
Seetting the queryType here seems a bit off I assume we could just inline it in the PrometheusQuery constructor?

Reply:
`queryType` can be (and will be in incoming PRs) `Range`, `Instant` and `Exemplars`. In query coming from frontend, we have `instant: boolean, range: boolean, exemplar: boolean` parameter, so it is not as easy as use  PrometheusQuery constructor. What I want to do now is create `queryType` variable that can be then used as a switch to run each of these queries.